### DISCUSSION TITLE
Continuous vulnerability scanning

### DISCUSSION THREAD

**Discussion by pbak95** [aquasecurity/trivy-operator#48] [Development]
Hello evryone!

I really enjoy this initiative since it really looks like sth I was thinking about from some time. I read through docs and maybe have an idea about feature proposal:

_As a **security engineer** I would like to have **an option to configure periodic vulnerability scanning** so that I can **be sure that my existing kubernetes workloads do not have any new vulnerabilities**._

Such feature should cover the case when workload is running on cluster from some time and in meantime new vulnerability **X** is discovered in one of it's dependencies. Vulnerability **X** wasn't known at the time workload was deployed on cluster.

What do you think about this? Sorry if this feature is already in place and I missed this part in documentation.

**Comment by chen-keinan** [comment]
@pbak95 we have such functionality, check this config [OPERATOR VULNERABILITY SCANNER REPORT TTL](https://github.com/aquasecurity/trivy-operator/blob/ba9a048c27e714c6821312bae1eba59f1a71ab7f/docs/operator/configuration.md)
It basically allows you to define the TTL for each vulnerability report; once its expires, the report gets deleted and a new scan is triggered for that specific workload which is associated to deleted report.

Let me know if it satisfied your request.

**Reply by pbak95** [reply]
@chen-keinan it should cover my case, thank you for help!
